db/httpdb.js

```diff
@@ -439,6 +440,54 @@ module.exports = function (
     )
   }
 
+  DB.prototype.lockAccount = function (account) {
+    var unlockCode = crypto.randomBytes(16).toString('hex');
+    log.trace({ op: 'DB.lockAccount', uid: account && account.uid, unlockCode: unlockCode })
+
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/lock',
+      {
+        lockedAt: Date.now(),
+        unlockCode: unlockCode
+      }
+    )
+  }
+
+  DB.prototype.unlockAccount = function (account) {
+    log.trace({ op: 'DB.unlockAccount', uid: account && account.uid })
+    return this.pool.post(
+      '/account/' + account.uid.toString('hex') + '/unlock'
+      )
+      .then(null,
+        function (err) {
+          if (err.statusCode === 404) {
+            // TODO - throw a better error here.
+            err = error.unknownAccount()
+          }
+          throw err
+        }
+      )
+  }
+
+  DB.prototype.unlockCode = function (account) {
+    log.trace({ op: 'DB.unlockCode', uid: account && account.uid })
+    return this.pool.get(
+      '/account/' + account.uid.toString('hex') + '/unlockCode'
+      )
+      .then(
+        function (body) {
+          return body.unlockCode
+        },
+        function (err) {
+          if (err.statusCode === 404) {
+            // TODO - throw a better error here.
+            err = error.unknownAccount()
```

Comment:
It seems like there are two possible reasons for a 404 here:

1. the account does not exist.
1. the account exists, but has no unlock code.

The DB drivers return a 404 in both cases. It would be nice to return `unknownAccount` when the account does not exist and `accountNotLocked` when the account exists, but is not locked.